src/python/pants/backend/python/goals/lockfile.py

```diff
@@ -269,44 +277,85 @@ async def generate_lockfiles_goal(
     return GenerateLockfilesGoal(exit_code=0)
 
 
+class AmbiguousResolveNamesError(Exception):
+    def __init__(self, ambiguous_names: list[str]) -> None:
+        if len(ambiguous_names) == 1:
+            first_paragraph = (
+                "The same resolve name is used both for a tool lockfile and a lockfile for your "
+                f"own code, so it's ambiguous what the resolve refers to: {ambiguous_names[0]}"
```

Comment:
Since users are unlikely to change the names of tool resolves, these errors could probably be more user focused:
```
A resolve name from $option collides with the name of a tool resolve.
```
... rather than referring to ambiguity...?